lib/node_modules/@stdlib/math/base/special/log1pf/test/test.js

```diff
@@ -0,0 +1,335 @@
+	var v;
+	var x;
+	var i;
+	var e;
+
+	x = mediumPositive.x;
+	expected = mediumPositive.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = log1pf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		delta = absf( v - e );
+		tol = EPS * absf( e );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + e + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `ln(x+1)` for negative small numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var v;
+	var x;
+	var i;
+	var e;
+
+	x = smallNegative.x;
+	expected = smallNegative.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = log1pf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		delta = absf( v - e );
+		tol = EPS * absf( e );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + e + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `ln(x+1)` for positive small numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var v;
+	var x;
+	var i;
+	var e;
+
+	x = smallPositive.x;
+	expected = smallPositive.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = log1pf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		delta = absf( v - e );
+		tol = EPS * absf( e );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + e + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `ln(x+1)` for negative tiny numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var v;
+	var x;
+	var i;
+	var e;
+
+	x = tinyNegative.x;
+	expected = tinyNegative.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = log1pf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		delta = absf( v - e );
+		tol = EPS * absf( e );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + e + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `ln(x+1)` for positive tiny numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var v;
+	var x;
+	var i;
+	var e;
+
+	x = tinyPositive.x;
+	expected = tinyPositive.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = log1pf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		delta = absf( v - e );
+		tol = EPS * absf( e );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + e + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function accurately computes `ln(x+1)` for negative large numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var v;
+	var x;
+	var i;
+	var e;
+
+	x = largeNegative.x;
+	expected = largeNegative.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = log1pf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		delta = absf( v - e );
+		tol = 1.0e07 * EPS * absf( e );
```

Comment:
```suggestion
		tol = 1.0e07 * EPS * absf( e );
```
The JS implementation requires a very large tolerance for numbers in the range `-0.9999` to `-0.41422`. But, for the C implementation, the required tolerance equals `EPS * absf( e )` only. I checked the implementation from the FreeBSD source, but couldn't find the reason.

FreeBSD reference: https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/s_log1pf.c?revision=367086&view=markup

Reply:
Yeah, that is a very large tolerance and would indicate that something is off with the implementation.